src/solver/optimisation/constraints/BindingConstraintHour.cpp

```diff
@@ -2,16 +2,25 @@
 
 void BindingConstraintHour::add(int pdt, int cntCouplante)
 {
-    data.CorrespondanceCntNativesCntOptim[pdt]
-      .NumeroDeContrainteDesContraintesCouplantes[cntCouplante]
-      = builder.data.nombreDeContraintes;
-
     const CONTRAINTES_COUPLANTES& MatriceDesContraintesCouplantes
       = data.MatriceDesContraintesCouplantes[cntCouplante];
-
+    // Are we dealing with an hourly binding constraint ?
     if (MatriceDesContraintesCouplantes.TypeDeContrainteCouplante != CONTRAINTE_HORAIRE)
         return;
 
+    // If so, is it enabled at this hour ?
+    if (!MatriceDesContraintesCouplantes.enabledAtHour[pdt])
+    {
+        // By convention, any value that is < 0 represents a non-existing constraint
+        data.CorrespondanceCntNativesCntOptim[pdt]
+          .NumeroDeContrainteDesContraintesCouplantes[cntCouplante] = -1;
+        return;
+    }
+
+    data.CorrespondanceCntNativesCntOptim[pdt]
+      .NumeroDeContrainteDesContraintesCouplantes[cntCouplante]
+      = builder.data.nombreDeContraintes;
+
```

Comment:
The trick is here. Where `NumeroDeContrainteDesContraintesCouplantes[cntCouplante] = -1` is interpreted as "no constraint". This is a general convention used whenever some constraint may or may not be enabled.